ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/clientlibs/site/js/fileinputwidget.js

```diff
@@ -163,7 +163,7 @@ class FileInputWidget {
             } else {
                 // todo: add support here
                 //let previewFileObjIdx = this._getFileObjIdx(index);
-                let previewFile = this.#widget.files[index];
+                let previewFile = this.#fileArr[index] ? this.#fileArr[index].data : this.#fileArr[index];
```

Comment:
if `this.#fileArr[index]` is undefined, then previewFile would always be underfined ?

Reply:
But the code is not readable, please change it.

Reply:
okay